services/hexathons/src/models/location.ts

```diff
@@ -0,0 +1,28 @@
+import { AccessibleRecordModel, accessibleRecordsPlugin } from "@casl/mongoose";
+import mongoose, { Schema, model, Types } from "mongoose";
+
+import { HexathonModel } from "./hexathon";
+
+export interface Location extends mongoose.Document {
+  hexathon: Types.ObjectId;
+  name: string;
+}
+
+const locationSchema = new Schema<Location>({
+  hexathon: {
+    type: Schema.Types.ObjectId,
+    required: true,
+    ref: HexathonModel,
+  },
+  name: {
+    type: String,
+    required: true,
+  },
```

Comment:
Locations don't need to have a relevant hackathon, usually we just keep the same locations across all events.